### PR TITLE
Allow disabling RStudio auth by setting blank password

### PR DESCRIPTION
**Pull request by JerryLeiDing**
If no password is provided, set the auth-none flag

### CODE REVIEW

File: rstudio/rstudio.sh [hunk 1]
```diff
@@ -68,9 +68,11 @@ function get_apt_key_for_debian() {
 }
 
 if [[ "${ROLE}" == 'Master' ]]; then
-  if (( ${#USER_PASSWORD} < 7 )) ; then
-    echo "You must specify a password of at least 7 characters for user \`$USER_NAME\` through metadata \`rstudio-password\`."
-    exit 1
+  if [[ ! -z ${USER_PASSWORD} ]] ; then
+    if (( ${#USER_PASSWORD} < 7 )) ; then
```
Comment (medb) [inline rstudio/rstudio.sh]: Merge 2 if's in single statement with `&&`?

File: rstudio/rstudio.sh [hunk 2]
```diff
@@ -68,9 +68,11 @@ function get_apt_key_for_debian() {
...
+      echo "You must specify a password of at least 7 characters for user \`$USER_NAME\` through metadata \`rstudio-password\`."
```
Comment (medb) [inline rstudio/rstudio.sh]: Maybe use regular single quotes instead, to avoid escaping of back-quotes?

File: rstudio/rstudio.sh [hunk 3]
```diff
@@ -68,9 +68,11 @@ function get_apt_key_for_debian() {
 }
 
 if [[ "${ROLE}" == 'Master' ]]; then
-  if (( ${#USER_PASSWORD} < 7 )) ; then
-    echo "You must specify a password of at least 7 characters for user \`$USER_NAME\` through metadata \`rstudio-password\`."
-    exit 1
+  if [[ ! -z ${USER_PASSWORD} ]] ; then
```
Comment (medb) [inline rstudio/rstudio.sh]: Use `-n` check of negating `-z` check

File: rstudio/rstudio.sh [hunk 4]
```diff
@@ -103,6 +105,13 @@ if [[ "${ROLE}" == 'Master' ]]; then
   fi
   if ! [ $(id -u "${USER_NAME}") ]; then
     useradd --create-home --gid "${USER_NAME}" "${USER_NAME}"
-    echo "${USER_NAME}:${USER_PASSWORD}" | chpasswd
+    if [[ ! -z "${USER_PASSWORD}" ]] ; then
```
Comment (medb) [inline rstudio/rstudio.sh]: Use `-n` instead of negating `-z` check

File: rstudio/rstudio.sh (hunk 1 above)
Comment (JerryLeiDing) [reply rstudio/rstudio.sh]: Done

File: rstudio/rstudio.sh (hunk 2 above)
Comment (JerryLeiDing) [reply rstudio/rstudio.sh]: Done

File: rstudio/rstudio.sh (hunk 3 above)
Comment (JerryLeiDing) [reply rstudio/rstudio.sh]: Done

File: rstudio/rstudio.sh (hunk 4 above)
Comment (JerryLeiDing) [reply rstudio/rstudio.sh]: Done

**Pull request merged**